### PR TITLE
Re-expose a simplified version of the stretch properties in SubViewport

### PR DESCRIPTION
**Pull request by YeldhamDev**
- Re-exposed a more simplified version of the `stretch` properties. The size override now takes effects always when both width and height are above `0`, discarding the argument to explicitly enable it. The `margin` argument was also discarded.
- Made `size` a member.
- Re-added some docs of the elements from the old `Viewport`.

### CODE REVIEW

File: scene/main/viewport.cpp [hunk 1]
```diff
@@ -3624,18 +3628,40 @@ void SubViewport::set_use_arvr(bool p_use_arvr) {
...
 
 void SubViewport::set_size(const Size2i &p_size) {
-	_set_size(p_size, Size2i(), Rect2i(), Transform2D(), true);
+	_set_size(p_size, _get_size_override(), Rect2i(), _stretch_transform(), true);
 }
 Size2i SubViewport::get_size() const {
 	return _get_size();
 }
 
+void SubViewport::set_size_override(const Size2i &p_size) {
```
Comment (reduz) [inline scene/main/viewport.cpp:3642]: I would rename to:

set_2d_size_override
set_2d_size_override_stretch

which I believe is closer to what it actually does.

**Review by reduz — CHANGES_REQUESTED** [review]

File: scene/main/viewport.cpp (hunk 1 above)
Comment (Calinou) [reply scene/main/viewport.cpp:3642]: @reduz Doesn't this also affect 3D rendering (i.e. lower sizes will make it pixelated)?

Comment (YeldhamDev) [reply scene/main/viewport.cpp:3642]: @Calinou Nope, it only affects 2D. I'm going to start making the changes.

**Comment by YeldhamDev** [comment]
@reduz Changes made. I named it `size_2d_override` as to not start it with a number.

**Review by reduz — APPROVED** [review]

**Pull request merged**